### PR TITLE
Deliver better speed than patrun on find/lookup

### LINKED ISSUE #44: is there any conditions then bloomrun.lookup faster than patrun.find?
`A js pattern matcher based on bloom filters, inspired by patrun. But different: 3-10x faster`

node v7.0.0

```
const Benchmark = require('benchmark')
const suite = new Benchmark.Suite

const patrun = require('patrun')({ gex: true })
const bloomrun = require('bloomrun')()

patrun.add({ role: 'test', action: 'other', id: '*' }, 'result')
bloomrun.add({ role: 'test', action: 'other', id: /.*/ }, 'result')

suite.add('patrun#add', () => {
  patrun.add({test: 'pattern'}, 'test pattern')
})
suite.add('bloomrun#add', () => {
  bloomrun.add({test: 'pattern'}, 'test pattern')
})

suite.add('patrun#find', () => {
  patrun.find({ role: 'test', action: 'other', id: 'qwe', payload: { qwe: 'asd', zxc: 'xxx' } })
})
suite.add('bloomrun#find', () => {
  bloomrun.lookup({ role: 'test', action: 'other', id: 'qwe', payload: { qwe: 'asd', zxc: 'xxx' } })
})

suite.on('cycle', event => console.log(String(event.target))).run({ 'async': true })
```

```
patrun#add x 332,083 ops/sec ±2.70% (76 runs sampled)
bloomrun#add x 437,241 ops/sec ±4.89% (71 runs sampled)
patrun#find x 1,074,769 ops/sec ±2.89% (73 runs sampled)
bloomrun#find x 309,006 ops/sec ±1.96% (77 runs sampled)
```

### PR DESCRIPTION
**Pull request by mcollina**
Fixed #44.

Removes regexp support on object passed to `lookup` and `list`, it would have been impossible to deliver that with good perf.

@mcdonnelldean @davidmarkclements can you please review?

@mcdonnelldean can you please update the logo, removing the bloomfilter part?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -192,9 +191,6 @@ This project was kindly sponsored by [nearForm](http://nearform.com).
 This library is heavily inspired by Richard Rodger's
 [patrun](http://npm.im/patrun) and [Seneca](http://npm.im/seneca).
 
-Bloomrun would not be possible without
-[bloomfilter](https://www.npmjs.com/package/bloomfilter).
-
 The bloomrun logo was created, with thanks, by [Dean McDonnell](https:/github.com/mcdonnelldean)
```
Comment (davidmarkclements) [inline README.md]: efforts made to use "BloomRun" - but not here - should this be "The BloomRun logo...."

File: README.md [hunk 2]
```diff
@@ -192,9 +191,6 @@ This project was kindly sponsored by [nearForm](http://nearform.com).
 This library is heavily inspired by Richard Rodger's
 [patrun](http://npm.im/patrun) and [Seneca](http://npm.im/seneca).
 
-Bloomrun would not be possible without
```
Comment (davidmarkclements) [inline README.md]: does this mean bloomrun doesn't use bloom filters? should the name be changed?

File: benchmarks/list.js [hunk 3]
```diff
@@ -162,7 +145,6 @@ var run = bench([
   threeEntries,
   fiveHundredEntries,
   fiveHundredEntriesAndProperties,
-  fiveHundredEntriesAndKnownProperties,
   patrunThreeEntries,
   patrunFiveHundredEntriesAndProperties
 ], 100000)
```
Comment (davidmarkclements) [inline benchmarks/list.js]: is it worth adding some deeper patterns, nesting 3 - 4 -5 levels deep?

**Comment by davidmarkclements** [comment]
overall seems good to me - one thing I want to check, does removing regexps for lookup and list affect mu (I don't think so from memory since mu special cases the '*') 

I would squash the commits, but up to you 

👍

File: README.md (hunk 2 above)
Comment (mcollina) [inline README.md]: Weeelll I would keep it as it is. It uses most of the same logic (buckets).

File: benchmarks/list.js (hunk 3 above)
Comment (mcollina) [inline benchmarks/list.js]: What do you mean by deep? The number of properties? Patrun does not offer multiple nested levels of objects. We did, but it was slowing down everything to a crawl (no inlining). Also, they make the software brittle IMHO, so they should not be used in the first place.

**Comment by mcollina** [comment]
In mu, you use regexps for mu.define, which are still supported. They are not supported anymore for mu.disparch

**Comment by davidmarkclements** [comment]
Cool, lgtm

File: README.md (hunk 1 above)
Comment (mcdonnelldean) [inline README.md]: @davidmarkclements I've slapped him on the wrist for this. It is bloomrun or Bloomrun if it must be capitalised, never BloomRun

**Comment by mcdonnelldean** [comment]
@mcollina All assets now updated to remove reference to bloomfilters.

**Comment by eljefedelrodeodeljefe** [comment]
API stays the same, but underlying mechanism have changed, hence semver major? @thekemkid, can you update our deps if possible? It's being marked by bithound.

**Comment by mcdonnelldean** [comment]
More due to removing deep patterns I believe.

**Comment by eljefedelrodeodeljefe** [comment]
okay. Just asking whether it's safe to update, but cannot see any problems for us. Just bumped the version. Thx.

**Comment by mcollina** [comment]
Most of the tests passed, only the ones on deep pattern matching were removed. The internal logic has changed heavily. Report any issues you might encounter!

**Pull request merged**